src/serialization/pointers/list/list.ts

```diff
@@ -263,37 +263,16 @@ export class List<T> extends Pointer implements Array<T> {
     return this;
   }
 
-  keys(): IterableIterator<number> {
-    const length = this.length;
-    return Array.from({ length }, (_, i) => i)[Symbol.iterator]();
+  keys(): ArrayIterator<number> {
```

Comment:
I'm not sure about these changes. It was needed to satisfy TS after the 5.6/5.7 upgrade, and it _seems_ to work

Reply:
Calling `toArray()`, causes a full (JS) copy and therefore stops two-way sync + custom iterator only generates entries until needed

Reply:
lets do it and iterate